builder/builder.go

```diff
@@ -642,3 +640,37 @@ func validateBuildkitEndpoint(ep string) (string, error) {
 	}
 	return ep, nil
 }
+
+// parseBuildkitFlags parses buildkit flags
+func parseBuildkitFlags(inp string, driver string) (res []string, err error) {
+	if inp != "" {
+		res, err = shlex.Split(inp)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse buildkit flags")
+		}
+	}
+
+	if driver == "kubernetes" || driver == "docker-container" {
+		var allowInsecureEntitlements []string
+		flags := pflag.NewFlagSet("buildkitd", pflag.ContinueOnError)
+		flags.Usage = func() {}
+		flags.StringArrayVar(&allowInsecureEntitlements, "allow-insecure-entitlement", nil, "")
+		_ = flags.Parse(res)
+
+		var hasNetworkHostEntitlement bool
+		for _, e := range allowInsecureEntitlements {
+			if e == "network.host" {
+				hasNetworkHostEntitlement = true
+				break
+			}
+		}
+
+		if !hasNetworkHostEntitlement {
+			// always set network.host entitlement as container network is
+			// isolated for docker-container and kubernetes drivers
```

Comment:
In the comment could you explain the purpose of setting the entitlement by default?